torch/distributed/_spmd/experimental_ops.py

```diff
@@ -17,17 +17,45 @@
 aten = torch.ops.aten  # pyre-ignore
 
 
-@register_prop_rule(aten._foreach_add.List)  # pyre-ignore
-def _prop__foreach_add(op_schema: OpSchema) -> OutputSharding:
-    self, other = op_schema.args_schema
+@register_prop_rule(  # pyre-ignore
+    [
+        aten._foreach_neg.default,
+        aten._foreach_reciprocal.default,
+        aten._foreach_sqrt.default,
+    ]
+)
+def _prop__foreach_unaop(op_schema: OpSchema) -> OutputSharding:
+    self = op_schema.args_schema[0]
     assert isinstance(self, list) and all(
         [isinstance(s, DTensorSpec) for s in self]
     )
+    # FIXME(@mrshenli): for sqrt, this is only mathematically correct for
+    # Replicate and Shard tensor.
```

Comment:
oh do you mean it needs to guard on partial?

Reply:
not sure yet, we might need to return arg schema suggestions with either Replicate or Shard to make this sound, but it doesn't seem the optimal decision on Replicate vs Shard can be make locally here?